Docker.DotNet.BasicAuth/MaybeSecureString.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Docker.DotNet.BasicAuth
+{
+    internal struct MaybeSecureString : IDisposable
+    {
+#if !netstandard
+        public SecureString Value;
+
+        public MaybeSecureString(string str)
+        {
+            var secureStr = new SecureString();
+
+            if (str.Length > 0)
+            {
+                foreach (char c in str)
+                {
+                    secureStr.AppendChar(c);
+                }
+            }
+
+            Value = secureStr;
+        }
+
+        public MaybeSecureString(SecureString value)
+        {
+            Value = value;
```

Comment:
Is the assumption here that the callers validate that value was not null? Seems like it might be better to have validation here and you cant say
```C#
new MaybeSecureString(null);
```